bundles/config/org.eclipse.smarthome.config.core/src/main/java/org/eclipse/smarthome/config/core/ConfigOptionProvider.java

```diff
@@ -28,8 +28,7 @@
      *            the parameter name for which the requested options shall be returned
      * @param locale
      *            locale
-     * @return the configuration options provided by this provider (not
-     *         null, could be empty)
+     * @return the configuration options provided by this provider if any (may be null)
```

Comment:
So, can you add some information on which situations we should return an empty collection and on which one we should return null?
What makes the difference?

Can we also change the current implementations in the ESH repo to use a consistent logic.

E.g.
* the URI is not handled => null, otherwise empty or filled collection
* URI or param not handled => null, otherwise a collection that contains the parameter